source-code/sdk-js/src/helpers/ast.ts

```diff
@@ -39,28 +40,32 @@ const wrapWithCallExpression = (n: Node, identifier: string) => {
  * @param identifier The function identifier with which the declarator should be wrapped.
  * @returns A promise that resolves to the resulting ast or rejects if the searched for declarator can't be found.
  */
-export const wrapVariableDeclaration = (async (sourceAst, searchIdentifier, identifier) => {
-	const sourceAstClone = structuredClone(sourceAst)
-	let found = false
-	walk(sourceAstClone, {
-		enter(node) {
-			if (
-				node.type === "VariableDeclarator" &&
-				node.id.type === "Identifier" &&
-				node.id.name === searchIdentifier &&
-				node.init
-			) {
-				found = true
-				const newNode = structuredClone(node)
-				newNode.init = wrapWithCallExpression(node.init, identifier)
-				this.replace(newNode)
-				this.skip()
-			}
-		},
-	})
-	if (!found) throw new Error("Couldn't find variable declarator.")
-	return sourceAstClone
-}) satisfies wrapVariableDeclaration
+export const wrapVariableDeclaration = ((sourceAst, searchIdentifier, identifier) => {
+	try {
```

Comment:
why do you wrap it in a try-catch-block? You can also just return the error. Or can the `walk` function also throw?

Reply:
It was very convenient as I am able to break the execution of the `walk` function by throwing an Error in there. So yes, the walk function can throw. But you are right, I don't need it in `WrapWithDeclaration`. Changing it there.